packages/build/src/plugins/compatibility.js

```diff
@@ -52,6 +60,27 @@ const getCompatibleEntry = async function ({ versions, nodeVersion, packageJson,
   return compatibleEntry || { version: versions[0].version }
 }
 
+// Takes an array of versions and an object containing all the enabled feature
+// flags. It finds the first entry in versions that contains a `featureFlag`
+// field that matches one of the enabled feature flags — if one is found, that
+// version entry is bumped to the first slot, which effectively makes it the
+// new default version; if none is found, the default version is not changed.
+// Either way, any versions with a `featureFlag` field are filtered out before
+// returning, so that normal evaluation of conditions can take place downstream
+// without any concern for feature flags.
+const filterVersionsByFeatureFlag = ({ featureFlags = {}, versions }) => {
+  const match = versions.find(({ featureFlag }) => featureFlag !== undefined && featureFlags[featureFlag] === true)
+  const versionsWithoutFeatureFlags = versions.filter(({ featureFlag }) => featureFlag === undefined)
```

Comment:
If we remove the bump to the first slot (see https://github.com/netlify/build/pull/3304/files#r678472517), we might be able to simplify this function to only:

```js
const filterVersionsByFeatureFlag = ({ featureFlags = {}, versions }) => {
  return versions.filter(({ featureFlag }) => featureFlag === undefined || featureFlags[featureFlag] === true)
}
```